components/MapComponent.tsx

```diff
@@ -0,0 +1,174 @@
+        In order to make this work, have an "initial zoom level" that's set at some point
+        Only allow zooms to be smaller than it, and when they are, make button appear
+        Button will set zoom to default level
+        Return the latitudeDelta and longitudeDelta to the original zoom level, save this level
+
+      Ideally pinch to zoom? zoomEnabled true
+      Join event, but you exit the area of the event
+      Zoom control default? zoomControlEnabled false, allow for the "Reset zoom" button, since goes in same spot
+      Getting userLocation would work, but would probably want to have marker for current location? Not sure. showsUserLocation might need to be on
+        If follows, then add in stuff where there's a certain "box" of the map that's visible
+        Find the maximum of the map, showing visible
+      Refresh map button might be useful. showsMyLocationButton might want to exist to be able 
+        Alternate: onRegionChangeComplete event data to refresh events that are shown when moving
+      Cannot implement "limit" to the map, using react-native-maps
+        Can however, implement onPanDrag to check if they drag to a certain part that's too far
+        Will either:
+            Refresh map to user location (might seem like error? prolly insert a brief text blurb)
+            Refresh map to nearby the border (might be confusing)
+
+
+
+export default MapComponent(){
+    return (
+    <View>
+
+    ------  MAP VIEW GOES DOWN HERE ---------
+        <MapView 
+            ref={mapRef}
+            initialRegion={userArea}
+            provider="google"
+
+        >
+        NEED TO HAVE DATA FROM EVENTS, CREATE CIRCLES TO REPRESENT EVENTS
+        GET DATA TO PLACE CIRCLES, BASED ON EVENTS
+        Events are a different kind of type from posts
+
+        Create a fake EventFeed elsewhere to supply information regarding events, that I would fetch
+        Allows for testing grabbing the info 
+        grab LatLngs, make them into circles 
+
+        FOR NOW, PUT TEST DATA
+        <Circle center={Lat} 
+
+
+        <MapView />
+
+
+    ------  MAP VIEW ENDS FROM HERE ---------
+
+    </View>
+
+    )
+}
+
+ */
+
+import React from "react";
+import { Dimensions, StyleSheet, View } from "react-native";
+import MapView, { Marker, PROVIDER_GOOGLE } from "react-native-maps";
+
+//Height information: Gets how tall/wide the device in use is
+const deviceHeight= Dimensions.get("window").height
+const deviceWidth= Dimensions.get("window").width
+
+//Placeholder data to figure out how markers work; objects seem to work
+const state = { markers: [
+  {
+    "key": 1,
+    "place": "Place one",
+    "lat": 34.079761,
+    "lng": -118.296802,
+  },
+  {
+    "key": 2,
+    "place": "Place two",
+    "lat": 34.039761,
+    "lng": -118.256802,
+  },
+]
+}
+
+//IDK PUT THE LOCATION YOU FETCH FROM THE LOCATION SERVICES HERE? Might be different, could just be placeholder
```

Comment:
@mhayes853 Will device location be included in your dependencies context? We do have a useLocation hook that will give you the device location, but we can worry about implementing that here in the next sprint.

Reply:
Yeah, we'll probably have some interface called `UserLocationClient` or something along those lines. We can then have a method on the interface which returns an [`AsyncIterable`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/asyncIterator) that we can use to observe the user's location in real time inside say a `useEffect`.